src/test/java/unit/WorkspaceUser.java

```diff
@@ -0,0 +1,191 @@
+package unit;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import bio.terra.cli.serialization.userfacing.UFWorkspaceUser;
+import bio.terra.workspace.model.IamRole;
+import com.fasterxml.jackson.core.type.TypeReference;
+import harness.TestCommand;
+import harness.TestUsers;
+import harness.baseclasses.SingleWorkspaceUnit;
+import java.io.IOException;
+import java.util.List;
+import java.util.Optional;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+/** Tests for the `terra workspace` commands that handle sharing with other users. */
+@Tag("unit")
+public class WorkspaceUser extends SingleWorkspaceUnit {
+  @BeforeEach
+  @Override
+  protected void setupEachTime() throws IOException {
+    workspaceCreator.login();
+
+    // `terra workspace set --id=$id`
+    TestCommand.runCommandExpectSuccess("workspace", "set", "--id=" + getWorkspaceId());
+
+    // `terra workspace list-users --format=json`
+    List<UFWorkspaceUser> listWorkspaceUsers =
+        TestCommand.runCommandExpectSuccess(
+            new TypeReference<>() {}, "workspace", "list-users", "--format=json");
```

Comment:
The `new TypeReference<>() {}` looks kind of weird. Maybe we should have a runCommandExpectSuccessList for List<T>, but maybe that's overkill.

Reply:
OK going to hold off on this for now.